zerver/tests/test_import_export.py

```diff
@@ -480,91 +480,89 @@ def test_import_realm(self) -> None:
-            for realm in realms]
-        self.assertEqual(emails[0], emails[1])
-        emails = [
-            {user.email for user in realm.get_active_users().filter(is_bot=False)}
-            for realm in realms]
-        self.assertEqual(emails[0], emails[1])
+        assert_realm_values(
+            lambda r: {user.email for user in r.get_admin_users()}
+        )
+
+        assert_realm_values(
+            lambda r: {user.email for user in r.get_active_users()}
+        )
 
         # test stream
-        stream_names = [
-            {stream.name for stream in get_active_streams(realm)}
-            for realm in realms]
-        self.assertEqual(stream_names[0], stream_names[1])
+        assert_realm_values(
+            lambda r: {stream.name for stream in get_active_streams(r)}
+        )
 
         # test recipients
-        stream_recipients = [
-            get_stream_recipient(Stream.objects.get(name='Verona', realm=realm).id)
-            for realm in realms]
-        self.assertEqual(stream_recipients[0].type, stream_recipients[1].type)
+        def get_recipient_stream(r: str) -> Stream:
+            return get_stream_recipient(
+                Stream.objects.get(name='Verona', realm=r).id
+            )
+
+        def get_recipient_user(r: str) -> UserProfile:
+            return get_personal_recipient(
+                UserProfile.objects.get(full_name='Iago', realm=r).id
+            )
 
-        user_recipients = [
-            get_personal_recipient(UserProfile.objects.get(full_name='Iago', realm=realm).id)
-            for realm in realms]
-        self.assertEqual(user_recipients[0].type, user_recipients[1].type)
+        assert_realm_values(lambda r: get_recipient_stream(r).type)
+        assert_realm_values(lambda r: get_recipient_user(r).type)
 
         # test subscription
-        stream_subscription = [
-            Subscription.objects.filter(recipient=recipient)
-            for recipient in stream_recipients]
-        subscription_stream_user = [
-            {sub.user_profile.email for sub in subscription}
-            for subscription in stream_subscription]
-        self.assertEqual(subscription_stream_user[0], subscription_stream_user[1])
-
-        user_subscription = [
-            Subscription.objects.filter(recipient=recipient)
-            for recipient in user_recipients]
-        subscription_user = [
-            {sub.user_profile.email for sub in subscription}
-            for subscription in user_subscription]
-        self.assertEqual(subscription_user[0], subscription_user[1])
+        def get_users(r: str) -> List[Set[str]]:
+            recipients = [get_recipient_stream(r), get_recipient_user(r)]
+            users = []
+            for recipient in recipients:
+                subscriptions = Subscription.objects.filter(recipient=recipient)
+                users.append({sub.user_profile.email for sub in subscriptions})
+            return users
+
```

Comment:
Ok, I saw your reply to my earlier comment, but I still think this is kind of confusing with these two-element arrays that are kinda more like tuples.

Let me take a stab...

Reply:
~~~ py
def get_subscribers(recipient):
    subscriptions = Subscription.objects.filter(recipient=recipient)
    users = {sub.user_profile.email for sub in subscriptions}
    return users

assert_realm_response(lambda r:
    get_subscribers(get_recipient_stream(r)))

assert_realm_response(lambda r:
    get_subscribers(get_recipient_users(r)))